tests/test_api.py

```diff
@@ -981,14 +985,140 @@ def test_empty_to_uuid(self):
             api.move_identity('03877f31261a6d1a1b3971d240e628259364b8ac', '')
 
 
+class TestAddOrganization(TestCase):
+    """Unit tests for add_organization"""
+
+    def test_add_new_organization(self):
+        """Check if everything goes OK when adding a new organization"""
+
+        organization = api.add_organization(name='Example')
+
+        # Tests
+
+        self.assertIsInstance(organization, Organization)
+
+        self.assertEqual(organization.name, 'Example')
+
+        organizations_db = Organization.objects.filter(name='Example')
+        self.assertEqual(len(organizations_db), 1)
+
+        org1 = organizations_db[0]
+
+        self.assertEqual(organization, org1)
+
+    def test_add_duplicate_organization(self):
+        """Check if it fails when adding a duplicate organization"""
+
+        org = api.add_organization(name='Example')
+
+        with self.assertRaisesRegex(AlreadyExistsError, ORGANIZATION_ALREADY_EXISTS_ERROR.format(name=org.name)):
+            org = api.add_organization(name=org.name)
+
+        organizations = Organization.objects.filter(name='Example')
+        self.assertEqual(len(organizations), 1)
+
+    def test_organization_name_none(self):
+        """Check if it fails when organization name is `None`"""
+
+        with self.assertRaisesRegex(InvalidValueError, ORGANIZATION_NAME_NONE_OR_EMPTY_ERROR):
+            api.add_organization(name=None)
+
+    def test_organization_name_empty(self):
+        """Check if it fails when organization name is empty"""
+
+        with self.assertRaisesRegex(InvalidValueError, ORGANIZATION_NAME_NONE_OR_EMPTY_ERROR):
+            api.add_organization(name='')
+
+    def test_organization_name_whitespaces(self):
+        """Check if it fails when organization name is composed by whitespaces only"""
+
+        with self.assertRaisesRegex(InvalidValueError, ORGANIZATION_NAME_NONE_OR_EMPTY_ERROR):
+            api.add_organization(name='   ')
+
+        with self.assertRaisesRegex(InvalidValueError, ORGANIZATION_NAME_NONE_OR_EMPTY_ERROR):
+            api.add_organization(name='\t')
+
+        with self.assertRaisesRegex(InvalidValueError, ORGANIZATION_NAME_NONE_OR_EMPTY_ERROR):
+            api.add_organization(name=' \t  ')
+
+    def test_organization_name_int(self):
+        """Check if it fails when organization name is an integer"""
+
+        with self.assertRaisesRegex(TypeError, ORGANIZATION_VALUE_ERROR):
+            api.add_organization(name=12345)
+
+
+class TestDeleteOrganization(TestCase):
+    """Unit tests for delete_organization"""
+
+    def setUp(self):
+        """Load initial dataset"""
+        api.add_organization(name='Example')
```

Comment:
Leave a blank line between the docstring and the next statement.

Reply:
Oops! Thank you for the heads-up.
Done.